src/core/thread/fiber.d

```diff
@@ -605,12 +605,12 @@ class Fiber
           size_t guardPageSize = PAGESIZE ) nothrow
     in
     {
-        assert( dg );
+        assert( cast(void delegate() const) dg );
```

Comment:
Why is this cast needed? It looks like a simple null check.

Reply:
It isn't a simple null check. It winds up calling dassert() which has overloads for delegates.

Reply:
That looks like an issue on its own, `assert(dg)` should just compile. Is there a bugzilla issue for that or am I missing something?

Reply:
assert's inside contracts behave differently.

Reply:
> assert's inside contracts behave differently.

Can you elaborate?

Reply:
Here's the error if I leave it unchanged:
```
src/core/thread/fiber.d(609): Error: none of the overloads of template `core.internal.dassert._d_assert_fail` are callable using argument types `!(void delegate())(string, void delegate())`
src/core/internal/dassert.d(38):        Candidates are: `_d_assert_fail(A)(scope const string op, auto ref scope const A a)`
src/core/internal/dassert.d(63):                        `_d_assert_fail(A...)`
```

Reply:
It looks like we should add that overload on druntime. Does it make sense to the user to cast it exolicitly to const?